docusaurus/docs/how-to-guides/updating-a-plugin.md

```diff
@@ -0,0 +1,73 @@
+---
+description: Learn how to use create-plugin update to automatically update configuration files, workflows, and dependencies.
+keywords:
+  - grafana
+  - plugin
+  - update
+  - config
+  - dependencies
+---
+
+import UpdateNPM from '@shared/createplugin-update.md';
+import UpdateNPMCommit from '@shared/createplugin-update-commit.md';
+import UpdateNPMForce from '@shared/createplugin-update-force.md';
+
+# Automate updating your plugin
+
+It's important for the health of your Grafana plugin to keep its tooling up to date. Doing so, however, can be a laborious task. To help solve this, `create-plugin` provides the `update` command to help you automate tooling update:
+
+- **configuration file changes** to take advantage of updates in Grafana, development tooling, and create-plugin
+- **dependency updates** to navigate major version bumps in development tooling
+- **code refactors** to align with changes to configuration files or major dependency updates
+
+:::info[Git branch status]
+
+Create a git branch before proceeding. If there are uncommitted changes, stash or commit them first otherwise the update command will exit early.
+
+:::
+
+To update your plugin, run:
+
+<UpdateNPM />
+
+After the update is finished, run the various development and build scripts to validate your plugin environment is still operating as expected.
+
+## Options and flags
+
+### `--commit`
+
+The commit flag will, on each successful migration, commit the changes to the current branch. This is useful for debugging and reviewing changes made by the update command.
```

Comment:
```suggestion
The commit flag will commit the changes to the current branch on each successful migration. This is useful for debugging and reviewing changes made by the update command.
```

Reply:
As in each successful migration is its own commit? I'm trying to wrap my head around how it helps with debugging

Reply:
Yes, each migration becomes a single commit.

Might just be me but if something's broken after running the command being able to look / move through commits to pinpoint where things broke is a lot easier than picking through all the changes without any context. Consider also that multiple migrations might have touched the same files multiple times for different reasons.